en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md

```diff
@@ -1,22 +1,26 @@
-# Floyd Cycle Detection Algorithm to find duplicate number in an array
+# Floyd Cycle Detection Algorithm to find duplicate numbers in an array
 
 ## Problem Statement
 
 Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
 
 ## Approach
-- We can imagine the array `arr` as a directed graph where each element is a node. Node with value `arr[i]` will point to node with value `arr[arr[i]]`.
-- The problem now becomes finding the node where two nodes with same value point towards.
-- Since a cycle is formed when two nodes with same value point towards the same node, we can use Floyd Cycle Detection Algorithm (also known as "Hare and Tortoise algorithm") to solve the problem.
-- The idea of algorithm is to maintain two pointers, `hare` and `tortoise` that iterate the array at different "speeds" (just like the fable). The details are as follow:
+- We can imagine the array `arr` as a directed graph where each element is a node. `arr[i]` is the index of the node to which the i-th node points.
+- For example, given the `arr = [1, 3, 4, 2, 3]`, we can represent `arr` as the following <br></br>
+![image](https://github.com/hollowcrust/Algorithms-Explanation/assets/72879387/a4afacb6-399b-44b6-96e9-b2fdfd75ae9d)
```

Comment:
You should add the image in this PR and use a relative link.

Reply:
Hi I uploaded the images in a separate folder and put the relative links in the markdown.